### ISSUE TITLE
dsc_host crashing on CentOS 8

### ISSUE THREAD

**Issue by brianloss** [microsoft/OMS-Agent-for-Linux#1282]
I have installed OMS Agent 1.13.33 on CentOS 8.2. The documentation indicates CentOS 8 is supported. However looking at the output for `dmesg`, I see segfault errors from dsc_host approximately every 15 minutes (there are several every 15 minutes). The messages look like the following:

```
[Tue Dec  8 22:58:05 2020] dsc_host[840579]: segfault at 28 ip 0000000000433fb4 sp 00007fff5089cdb0 error 4 in dsc_host[400000+94000]
[Tue Dec  8 22:58:05 2020] Code: 89 4c 24 30 64 48 8b 04 25 28 00 00 00 48 89 84 24 a8 00 00 00 31 c0 4d 85 e4 b0 04 0f 84 94 01 00 00 49 c7 04 24 00 00 00 00 <48> 8b 6f 28 49 89 fe 49 8b 50 18 48 8d 05 3e 19 04 00 49 89 cf 4c
```

**Comment by johanburati** [comment]
I'm seeing the same error on **Centos 8.1**, **Centos 8.2**, **RHEL 8.1** and **RHEL 8.2**:

```
kernel: dsc_host[9960]: segfault at 28 ip 0000000000433fb4 sp 00007ffc964cb850 error 4 in dsc_host[400000+94000]
kernel: Code: 89 4c 24 30 64 48 8b 04 25 28 00 00 00 48 89 84 24 a8 00 00 00 31 c0 4d 85 e4 b0 04 0f 84 94 01 00 00 49 c7 04 24 00 00 00 00 <48> 8b 6f 28 49 89 fe 49 8b 50 18 48 8d 05 3e 19 04 00 49 89 cf 4c
```
@brianloss did you open a support request for this ?

**Comment by brianloss** [comment]
@johanburati I thought this was the place to report issues. If there's somewhere else I should report, I'm happy to do so if you can point me in the right direction.

**Comment by johanburati** [comment]
It does not appear that the developers are actively monitoring [issues](https://github.com/microsoft/OMS-Agent-for-Linux/issues) on this project.

If you are having this issue with a specific VM, I would suggest you create a support request by clicking *New support request* at the bottom left menu on that VM blade in the **Azure Portal**:

![image](https://user-images.githubusercontent.com/106444/102702781-e1cac280-42a9-11eb-9c25-61039d979247.png)

If you do please let me know the number or tell the case owner to create a collab with **joburati**, I will follow up to fix this issue.

**Comment by johanburati** [comment]
I cut an issue in the other repo: https://github.com/microsoft/PowerShell-DSC-for-Linux/issues/764

**Comment by johanburati** [comment]
Putting the solution here as well so it is easier to find.

It's kind of the same thing as for RHEL 8 and other extensions: https://github.com/Azure/WALinuxAgent/issues/1719

## SOLUTION
Installing **python2**:
```
sudo dnf install python2 -y
sudo alternatives --set python /usr/bin/python2
```

Did not see any crash after doing that.

**Comment by Klaas-** [comment]
I think Microsoft needs to have a real look at all their python-dependent linux software. there is another issue #1292 that seems to have  a similar issue :)

Anyway I'll repeat here what I said in the other issue: This should be addressed by properly recognizing python and not by aliasing the python command. Red Hat does not recommend to set an unversioned python link ( see man unversioned-python ) on a recent rhel.

**Comment by brianloss** [comment]
@johanburati thanks for the follow-up. I kind of lost track of this while on break over the holidays, and didn't get around to filing a support case in Azure. Did you end up doing that, or should I still file it?

> Anyway I'll repeat here what I said in the other issue: This should be addressed by properly recognizing python and not by aliasing the python command. Red Hat does not recommend to set an unversioned python link ( see man unversioned-python ) on a recent rhel.

The release notes for 1.13.33 explicitly state that there is no longer a requirement to install python 2 or alias the python command to it (see [here](https://github.com/microsoft/OMS-Agent-for-Linux/releases/tag/OMSAgent_v1.13.33-0)). It would appear those notes are incorrect due to the "broken" dependency PowerShell-DSC-for-Linux.

**Comment by Klaas-** [comment]
@brianloss as a customer myself, I would suggest to file an issue with microsoft. That way microsoft can properly qualify the impact of an issue.

**Comment by johanburati** [comment]
@brianloss there are already a couple of support case open for this, it is true what I have submitted is more of a workaround, at least it will prevent the program from crashing and generating core files until the developers release a new version that fix the issue.

**Comment by johanburati** [comment]
@Klass- thanks for sharing the case number, I've sent you an update by email. 
@brianloss please check Klass's ticket notes, I have added the link for the internal ticket with the devs.

**Comment by 1lolbus1** [comment]
Thanks for the solution @johanburati ! 
I ended up using a symlink to the same effect
```
ln -s /usr/bin/python3 /usr/bin/python
```

**Comment by Klaas-** [comment]
This seems to be fixed by omsconfig-1.1.1-930.x86_64, you should no longer recommend the python workarounds in docs :)

**Comment by hestolz** [comment]
Thank you for your submission. As [previously announced](https://azure.microsoft.com/en-us/updates/were-retiring-the-log-analytics-agent-in-azure-monitor-on-31-august-2024/), the Log Analytics agent has been **deprecatated** and has no support as of **August 31, 2024.** If you use the Log Analytics agent to ingest data to Azure Monitor, [migrate now to the new Azure Monitor agent](https://docs.microsoft.com/en-us/azure/azure-monitor/agents/azure-monitor-agent-migration). As part of repo archival, open issues and pull requests will be closed.
